salt/cni/update-flannel-pillar.sh

```diff
@@ -0,0 +1,15 @@
+#!/bin/sh
+
+
+# set the "flannel:backend" to vxlan in the database iff no previous value was set
+logger -t "cni-migration" "persisting flannel:backend in database"
+velum_cont=$(docker ps | grep velum-dashboard | awk '{print $1}')
+docker exec "$velum_cont" entrypoint.sh bundle exec rails runner \
+       'ActiveRecord::Base.logger=nil ; \
+        found = Pillar.find_by(pillar: "flannel:backend") ; \
+        if found.nil? then \
+          puts "[CNI migration]: no previous value found for flannel:backend: setting vxlan" ; \
+          Pillar.create pillar: "flannel:backend", value: "vxlan" ; \
+        else \
+          puts "[CNI migration]: flannel:backend already set in database" ; \
```

Comment:
I think this code is not right.  In CaaS 1.0, `pillar.sls` will contain `flannel:backend = udp`, so we need to 1) get the pillar value from `params.sls` 2) save it to the database before starting the update. So the database value would have higher priority than the new `params.sls` file installed by the updated Salt code, and the `flannel:backend` would not be changed for existing clusters.

@ereslibre do you know how this could be done? `Pillar.find_by` does not seem to show values from the `params.sls` file...

Reply:
We can this in a slightly different approach as we have done in https://github.com/kubic-project/velum/pull/302/files#diff-322caad16c493d5f6eb76781540c1226 or https://github.com/kubic-project/velum/pull/303/files#diff-a8e35dc21156a5e0f22811a949f91ca9

These are migrations that will only be executed on 1.0 machines. On 2.0 installations they won't be executed if the schema file includes the version. So, the default pillar on salt should be set to whatever we want to be the default for the next versions, and for 1.0 installations, the migration adding the pillar override will be executed setting `flannel:backend` to `udp`.

So, new installations don't have this pillar override, while old installations have the pillar override `flannel:backend` = `udp`.